forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -7199,6 +7199,11 @@ public final boolean canBeSacrificedBy(final SpellAbility source, final boolean
             return false;
         }
 
+        // can't sacrifice it for mana ability if it is already marked as sacrifice
+        if (source != null && source.isManaAbility() && isUsedToPay()) {
+            return false;
+        }
```

Comment:
that i need to look at when i refactor the CostPayment in the far future

Reply:
yea I'm still not happy with isUsedToPay because it's tricky to reset correctly
but I will try to look into it later